packages/ampd-sdk/src/grpc/client.rs

```diff
@@ -0,0 +1,75 @@
+use async_trait::async_trait;
+use error_stack::{Report, Result};
+use mockall::automock;
+use thiserror::Error;
+use tonic::{codegen, transport, Streaming};
+
+use super::proto::blockchain_service_client::BlockchainServiceClient;
+use super::proto::crypto_service_client::CryptoServiceClient;
+use super::proto::{SubscribeRequest, SubscribeResponse};
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("failed to connect to the grpc endpoint")]
+    GRpcConnection(#[from] tonic::transport::Error),
+
+    #[error("failed to execute gRPC request")]
+    GRpcRequest(#[from] tonic::Status),
```

Comment:
Again, it's OK for now, but we will need to create more specific error for each status code later so that whoever using this SDK can decide what to do in each case.